napari/layers/image/image.py

```diff
@@ -483,21 +483,22 @@ def _set_view_slice(self):
                 scale[d] = self.downsample_factors[self.data_level][d]
             self._transforms['tile2data'].scale = scale
 
-            corner_pixels = np.clip(
-                self.corner_pixels,
-                0,
-                np.subtract(self.level_shapes[self.data_level], 1),
-            )
+            if self.dims.ndisplay == 2:
```

Comment:
Once I fixed `_update_multiscale`, I noticed that rotating a 3D cube resulted in the cube disappearing as it was rotated, until it was completely gone if looking at it side on. This change fixes it, though I don't have a test for it.